### PR TITLE
Tests for Aurora to Eth fast bridge

### PR DESCRIPTION
**Pull request by olga24912**
* **Double spend test:** two users make equal transaction, unlock tokens and one user try to withdraw tokens two times in a raw
* **Token transfer fail:** init_token_transfer fails because no storage deposit for the fast-bridge, init_token_transfer fails because of small amount of gas provided, 
* **Withdraw without withdraw from near:** two users init token transfer, both unlock tokens, try to withdraw without `withdraw_from_near`, first withdraw from near and the second withdraw without `withdraw_from_near`, the first user the second time `withdraw_from_near` and withdraw his tokens. 
* **Get Near address test:** check the correctness of the get near address function -- check that the returned address equal to the address in transactions
* **Whitelist mode test** check that whitelist mode works correctly. Enables and disables whitelist mode and checks the init_transfers possibility for different users.

### CODE REVIEW

File: aurora/integration-tests/src/lib.rs [hunk 1]
```diff
@@ -272,15 +307,67 @@ mod tests {
...
+            if let TransactionStatus::Succeed(res) = result.status {
+                let mut buf = [0u8; 20];
+                buf.copy_from_slice(&res.as_slice()[res.len() - 20..res.len()]);
+                return Some(buf);
             }
+
+            return None;
+        }
+
+        pub async fn get_near_address(&self) -> Option<String> {
```
[the remark this answers is not in the document]
Comment (olga24912) [reply aurora/integration-tests/src/lib.rs]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/340450b7a046472eea29678e631e0ad3e8cf6426

File: aurora/integration-tests/src/lib.rs [hunk 2]
```diff
@@ -129,20 +133,36 @@ mod tests {
...
+            user_account: Option<Account>,
+            check_output: bool,
+            gas: Option<u64>,
+        ) {
+            let valid_till = valid_till.unwrap_or(
+                (std::time::SystemTime::now()
+                    .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_nanos()
+                    + Duration::from_secs(30).as_nanos()) as u64,
```
[the remark this answers is not in the document]
Comment (olga24912) [reply aurora/integration-tests/src/lib.rs]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/be8dec1c5148da237503f2ba384a9bc25b2849ec

File: aurora/integration-tests/src/lib.rs [hunk 3]
```diff
@@ -32,10 +32,12 @@ mod tests {
     const NEAR_DEPOSIT: u128 = 2 * near_sdk::ONE_NEAR;
     const WNEAR_FOR_TOKENS_TRANSFERS: u128 = 100 * near_sdk::ONE_YOCTO;
 
+    const TRANSFER_EXPIRATION_PERIOD_S: u64 = 30;
```
Comment (karim-en) [inline aurora/integration-tests/src/lib.rs:35]:
```suggestion
    const TRANSFER_EXPIRATION_PERIOD_SEC: u64 = 30;
```

File: aurora/integration-tests/src/lib.rs [hunk 4]
```diff
@@ -129,20 +136,36 @@ mod tests {
...
+            user_account: Option<Account>,
+            check_output: bool,
+            gas: Option<u64>,
+        ) {
+            let valid_till = valid_till.unwrap_or(
+                (std::time::SystemTime::now()
+                    .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_nanos()
+                    + Duration::from_secs(TRANSFER_EXPIRATION_PERIOD_S).as_nanos()) as u64,
```
Comment (karim-en) [inline aurora/integration-tests/src/lib.rs:167]:
```suggestion
                    + Duration::from_secs(TRANSFER_EXPIRATION_PERIOD_SEC).as_nanos()) as u64,
```

**Review by karim-en — CHANGES_REQUESTED** [review]

File: aurora/integration-tests/src/lib.rs (hunk 3 above)
Comment (sept-en) [reply aurora/integration-tests/src/lib.rs:35]: IMO, the `_S` suffix is sufficient enough. E.g., if we use milliseconds, we would use `_MS`. But it's a matter of taste.

Comment (olga24912) [reply aurora/integration-tests/src/lib.rs]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/fc520f211b971e073aae04aa32c26491e606f6d5

File: aurora/integration-tests/src/lib.rs (hunk 4 above)
Comment (olga24912) [reply aurora/integration-tests/src/lib.rs]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/fc520f211b971e073aae04aa32c26491e606f6d5

File: aurora/integration-tests/src/lib.rs [hunk 5]
```diff
@@ -129,20 +133,36 @@ mod tests {
...
-        pub async fn init_token_transfer(&self, amount: u128, fee_amount: u128, valid_till: u64) {
+        pub async fn init_token_transfer(
+            &self,
+            amount: u128,
+            fee_amount: u128,
+            valid_till: Option<u64>,
+            user_address: Option<Address>,
+            user_account: Option<Account>,
+            check_output: bool,
+            gas: Option<u64>,
```
[the remark this answers is not in the document]
Comment (olga24912) [reply aurora/integration-tests/src/lib.rs]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/275f834d07e5281ac7b43ec61472ffd040fbdad0
https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/cf8c079c2467b2e367be17ee8e112ef8cab50be9
https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/384238884df2437a35e141dd762ae11e74183baf

**Review by karim-en — APPROVED** [review]

**Pull request merged**